spot-contracts/contracts/_interfaces/IBondIssuer.sol

```diff
@@ -31,4 +41,12 @@ interface IBondIssuer {
     /// @notice The bond address from the issued list by index.
     /// @return Address of the bond.
     function issuedBondAt(uint256 index) external view returns (IBondController);
+
+    /// @notice The total number of bonds issued by this issuer which are currently active.
+    /// @return Number of active bonds.
+    function activeCount() external view returns (uint256);
+
+    /// @notice The bond address from the active list by index.
+    /// @return Address of the active bond.
```

Comment:
This is actually a very weird interface definition, which makes me think our interface definition isn't quite right.

`index` is not the index of the bond in the bonds container. It's an index to an array that gives you an index into the bond container.

Also, `index` is an index into a datastructure that's not mentioned or documented anywhere else in the interface file. And the order of the indexes in this datastructure depends on the implementation, which isn't constrained here.

Does it need to be here? If so, does it make sense to have the activeBondIDX as part of the interface?

Reply:
The index of the bonds in the bonds container is private isn't exposed publicly. 

As far as the outside world is concerned, The there are `activeCount()` active bonds and they can be accessed by iterating from [0, activeCount()-1] (which are the indices of the active list).

It's a common practice to expose storage arrays through a `count()` method and a `read(i)` method ..

Reply:
Agree it's a little awkward. Do we have to expose this it all? `matureActive()` does not require an input, and emits the `BondMature(bond)` to log which bonds where matured anyway.

Reply:
Ah, in the full context I see the pattern of simulating an array. (The code review interface folds unchanged code above it out of view :-/ )
```
issuedCount()
issuedBondAt(i)
activeCount()
activeBondat(i)
```
This makes more sense, I'm ok with keeping it.

Reply:
Minor: The @notice lines should be @param lines in the natspec

Reply:
I think its a fair point if we want to expose it at all (since `matureActive` doesnt require an input). 

Leaning toward removing `activeCount` and `activeBondAt` from the interface definition but keeping it in the contract.